test/unit/test_utils.py

```diff
@@ -393,3 +393,19 @@ def test_smart_open_without_suffix():
 def test_compute_lengths(data, expected_lengths):
     lengths = utils.compute_lengths(mx.sym.Variable('data')).eval(data=data)[0]
     assert (lengths.asnumpy() == expected_lengths.asnumpy()).all()
+
+
+def test_read_metric_file():
```

Comment:
how about moving the logic for parsing a single metric file entry into a function and then just test that function, to avoid the temporary file for testing?

Reply:
Agree. Splitted it a separate function.